streaming/src/queue/transport.h

```diff
@@ -0,0 +1,256 @@
+              const ActorID &actor_id, const ActorID &peer_actor_id,
+              const std::string actor_handle_serialized, 
+              const std::vector<ObjectID> &queue_ids, const std::vector<ObjectID> &rescale_queue_ids,
+              std::string test_suite_name, std::string test_name,
+              uint64_t param)
+      : Message(actor_id, peer_actor_id, queue_ids[0]),
+  actor_handle_serialized_(actor_handle_serialized), queue_ids_(queue_ids), rescale_queue_ids_(rescale_queue_ids),
+  role_(role), test_suite_name_(test_suite_name), test_name_(test_name), param_(param) {}
+  virtual ~TestInitMsg() {}
+
+  static std::shared_ptr<TestInitMsg> FromBytes(uint8_t *bytes);
+  virtual void ConstructFlatBuf(flatbuffers::FlatBufferBuilder &builder);
+  queue::flatbuf::MessageType Type() { return type_; }
+  std::string ActorHandleSerialized() { return actor_handle_serialized_; }
+  queue::flatbuf::StreamingQueueTestRole Role() { return role_; }
+  std::vector<ObjectID> QueueIds() { return queue_ids_; }
+  std::vector<ObjectID> RescaleQueueIds() { return rescale_queue_ids_; }
+  std::string TestSuiteName() { return test_suite_name_; }
+  std::string TestName() { return test_name_;}
+  uint64_t Param() { return param_; }
+
+  std::string ToString() {
+    std::ostringstream os;
+    os << "actor_handle_serialized: " << actor_handle_serialized_;
+    os << " actor_id: " << ActorId();
+    os << " peer_actor_id: " << PeerActorId();
+    os << " queue_ids:[";
+    for (auto &qid : queue_ids_) {
+      os << qid << ",";
+    }
+    os << "], rescale_queue_ids:[";
+    for (auto &qid : rescale_queue_ids_) {
+      os << qid << ",";
+    }
+    os << "],";
+    os << " role:" << queue::flatbuf::EnumNameStreamingQueueTestRole(role_);
+    os << " suite_name: " << test_suite_name_;
+    os << " test_name: " << test_name_;
+    os << " param: " << param_;
+    return os.str();
+  }
+ private:
+  const queue::flatbuf::MessageType type_ =
+      queue::flatbuf::MessageType::StreamingQueueTestInitMsg;
+  std::string actor_handle_serialized_;
+  std::vector<ObjectID> queue_ids_;
+  std::vector<ObjectID> rescale_queue_ids_;
+  queue::flatbuf::StreamingQueueTestRole role_;
+  std::string test_suite_name_;
+  std::string test_name_;
+  uint64_t param_;
+};
+
+class TestCheckStatusRspMsg : public Message {
+ public:
+  TestCheckStatusRspMsg(const std::string test_name, bool status)
+      : test_name_(test_name), status_(status) {}
+  virtual ~TestCheckStatusRspMsg() {}
+
+  static std::shared_ptr<TestCheckStatusRspMsg> FromBytes(uint8_t *bytes);
+  virtual void ConstructFlatBuf(flatbuffers::FlatBufferBuilder &builder);
+  queue::flatbuf::MessageType Type() { return type_; }
+  std::string TestName() { return test_name_; }
+  bool Status() { return status_; }
+
+ private:
+  const queue::flatbuf::MessageType type_ =
+      queue::flatbuf::MessageType::StreamingQueueTestCheckStatusRspMsg;
+  std::string test_name_;
+  bool status_;
+};
+
+/// duplex
```

Comment:
This comment is very unclear. 
We could document more details, e.g., the purposes of responsibilities of this class, who will use this class, etc.

Also, will there be other subclasses besides `DirectCallTransport`?

Reply:
> This comment is very unclear.
> We could document more details, e.g., the purposes of responsibilities of this class, who will use this class, etc.
> 
> Also, will there be other subclasses besides `DirectCallTransport`?

In our internal version, we have another transport class for test, the inheritance can be removed here.

Reply:
> This comment is very unclear.
> We could document more details, e.g., the purposes of responsibilities of this class, who will use this class, etc.
> 
> Also, will there be other subclasses besides `DirectCallTransport`?

fixed.